Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -915,10 +915,10 @@ void doVayDepositionShapeN (const GetParticlePosition& GetPosition,
 
                 // Re-casting sx_new and sz_new from double to amrex::Real so that
                 // Atomic::Add has consistent types in its argument
-                amrex::Real const sxn_szn = sx_new[i] * sz_new[k];
-                amrex::Real const sxo_szn = sx_old[i] * sz_new[k];
-                amrex::Real const sxn_szo = sx_new[i] * sz_old[k];
-                amrex::Real const sxo_szo = sx_old[i] * sz_old[k];
+                amrex::Real const sxn_szn = static_cast<amrex::Real>(sx_new[i] * sz_new[k]);
+                amrex::Real const sxo_szn = static_cast<amrex::Real>(sx_old[i] * sz_new[k]);
+                amrex::Real const sxn_szo = static_cast<amrex::Real>(sx_new[i] * sz_old[k]);
+                amrex::Real const sxo_szo = static_cast<amrex::Real>(sx_old[i] * sz_old[k]);
```

Comment:
Avoids type repetition from the RHS:
```suggestion
                auto const sxn_szn = static_cast<amrex::Real>(sx_new[i] * sz_new[k]);
                auto const sxo_szn = static_cast<amrex::Real>(sx_old[i] * sz_new[k]);
                auto const sxn_szo = static_cast<amrex::Real>(sx_new[i] * sz_old[k]);
                auto const sxo_szo = static_cast<amrex::Real>(sx_old[i] * sz_old[k]);
```

Reply:
I can make this change if you insist, but my preference is to avoid using `auto` since it makes the code harder to follow (since extra mental effort is needed to parse the line and determine the type). The `auto` only seems to add obfuscation and doesn't seem to improve anything.

Reply:
I agree on not over-using `auto`. In the specific case here, it is useful to pick `auto`, because it is repetitive to write the RHS, which is ` = static_cast<amrex::Real>( ... );`.

Reply:
My old brain likes having the type explicit, and not need to do even that tiny but of extra work to see the cast.

What happens if someone later realizes a fix is needed and writes
`auto const sxo_szo = static_cast<amrex::Real>(sx_old[i] * sz_old[k])*2.;`
It seems that having the type explicit is more robust since it can be checked by the compiler.